addons/mail/static/src/core/common/thread.js

```diff
@@ -160,7 +160,13 @@ export class Thread extends Component {
                 if (this.env.chatter) {
                     this.env.chatter.fetchMessages = false;
                 }
-                toRaw(this.props.thread).fetchNewMessages();
+                if (this.props.thread.selfMember && this.props.thread.shouldScrollToUnread) {
+                    toRaw(this.props.thread).loadAround(
+                        this.props.thread.selfMember.new_message_separator
+                    );
+                } else {
+                    toRaw(this.props.thread).fetchNewMessages();
+                }
```

Comment:
`loadAround()` (no params) is equivalent to `fetchNewMessages()` I think (if not, would be nice this is the same).
We could improve shape of code by always calling `loadAround()` but only param is conditional.